pkg/operator/configobservation/images/observe_images.go

```
@@ -1,42 +1,163 @@
-		unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		err := unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
 	}
 
 	if !listers.ImageConfigSynced() {
 		glog.Warning("images.config.openshift.io not synced")
 		return prevObservedConfig, errs
 	}
 
+	// now gather the cluster config and turn it into the observed config
 	observedConfig := map[string]interface{}{}
 	configImage, err := listers.ImageConfigLister.Get("cluster")
 	if errors.IsNotFound(err) {
 		glog.Warningf("image.config.openshift.io/cluster: not found")
-		return observedConfig, errs
+		return observedConfig, append(errs, err)
 	}
 	if err != nil {
-		return prevObservedConfig, errs
+		return prevObservedConfig, append(errs, err)
 	}
+
 	internalRegistryHostName := configImage.Status.InternalRegistryHostname
 	if len(internalRegistryHostName) > 0 {
-		unstructured.SetNestedField(observedConfig, internalRegistryHostName, internalRegistryHostnamePath...)
+		err = unstructured.SetNestedField(observedConfig, internalRegistryHostName, internalRegistryHostnamePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	return observedConfig, errs
+}
+
+func ObserveExternalRegistryHostnames(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	prevObservedConfig := map[string]interface{}{}
+
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
+	externalRegistryHostnamePath := []string{"imagePolicyConfig", "externalRegistryHostnames"}
+	o, _, err := unstructured.NestedSlice(existingConfig, externalRegistryHostnamePath...)
```

accidentally ignoring this error?

---

yeah, will fix